### PR TITLE
[ONNX] Bump onnx submodule to release 1.13.1 from rc2

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96325

### CODE REVIEW

File: .ci/onnx/test.sh [hunk 1]
```diff
@@ -61,8 +61,7 @@ if [[ "$BUILD_ENVIRONMENT" == *onnx* ]]; then
   pip install -q --user --no-use-pep517 "git+https://github.com/pytorch/vision.git@$(cat .github/ci_commit_pins/vision.txt)"
   pip install -q --user transformers==4.25.1
   pip install -q --user ninja flatbuffers==2.0 numpy==1.22.4 onnxruntime==1.14.0 beartype==0.10.4
-  # TODO: change this when onnx 1.13.1 is released.
-  pip install --no-use-pep517 'onnx @ git+https://github.com/onnx/onnx@e192ba01e438d22ca2dedd7956e28e3551626c91'
+  pip install -q --user onnx==1.13.1
```
Comment (justinchuby) [inline .ci/onnx/test.sh:64]: combine this line with that on top to save some install time?

**Review by justinchuby — APPROVED** [review]

File: .ci/onnx/test.sh (hunk 1 above)
Comment (BowenBao) [reply .ci/onnx/test.sh:64]: I'll leave that to next time to not re-trigger another round of CI since it's marginal improvement.

**Pull request closed without merging**